### PR TITLE
feat: add Memo, KeyMemo

### PR DESCRIPTION
**Pull request by vanya2h**

### CODE REVIEW

File: packages/cache/src/memo.ts [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+	}
+
+	async get(force = false): Promise<T> {
+		if (force) {
+			this.clear()
+		}
+		const s = this.atom.get()
+		if (s.status === "rejected" || s.status === "idle") {
+			save(this._loader(), this.atom).then()
+		}
```
Comment (evgenynacu) [inline packages/cache/src/memo.ts:75]: Do we need these lines? It seems the last one is enough?

File: packages/react/src/use-rx.ts [hunk 2]
```diff
@@ -54,13 +55,11 @@ export function useRx<T>(observable: WrappedObservable<T>, deps: any[] = [observ
 						setCount(c => c + 1)
 					}
 				}
-				// eslint-disable-next-line react-hooks/exhaustive-deps
 			}),
-		deps
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+		[...deps, memoized]
```
Comment (evgenynacu) [inline packages/react/src/use-rx.ts:60]: Are you sure it's needed? It was intentionally not added there. It can break things I think

**Comment by evgenynacu** [comment]
Seems like a good implementation, simpler than my initial.
Added some comments. I think, there is no need to change Rx or useRx and this can be dangerous.
Also, is key memo implementation the same as cache memo or there any differences?

File: packages/cache/src/memo.ts (hunk 1 above)
Comment (vanya2h) [reply packages/cache/src/memo.ts:75]: yeah, you right. removed it

File: packages/react/src/use-rx.ts (hunk 2 above)
Comment (vanya2h) [reply packages/react/src/use-rx.ts:60]: reverted

**Comment by vanya2h** [comment]
> Seems like a good implementation, simpler than my initial. Added some comments. I think, there is no need to change Rx or useRx and this can be dangerous. Also, is key memo implementation the same as cache memo or there any differences?

There's only one difference between Cache and Memo - Memo doesn't have initial value
It'll behave just the same as we will just use async observable

**Pull request merged**